### PR TITLE
Moving to React Native 0.73.6

### PR DESCRIPTION
**Pull request by wmathurin**
I ran react-native cli to create a new app from a template then compared the output (package.json, podfile etc).

### CODE REVIEW

File: libs/SalesforceReact/package.json [hunk 1]
```diff
@@ -8,31 +8,28 @@
   "dependencies": {
     "create-react-class": "^15.7.0",
     "react": "18.2.0",
-    "react-native": "0.72.7",
-    "react-native-force": "git+https://github.com/forcedotcom/SalesforceMobileSDK-ReactNative.git#dev"
+    "react-native": "0.73.6",
+    "react-native-force": "git+https://github.com/wmathurin/SalesforceMobileSDK-ReactNative.git#rn73"
```
Comment (wmathurin) [inline libs/SalesforceReact/package.json:12]: Pointing to wmathurin rn73 to allow tests to pass. Will revert that line before merging.

**Review by brandonpage — APPROVED** [review]

**Comment by brandonpage** [comment]
@wmathurin Build is failing due to: `Error: Cannot find module 'metro-react-native-babel-preset'`

File: libs/SalesforceReact/package.json (hunk 1 above)
Comment (wmathurin) [reply libs/SalesforceReact/package.json:12]: Locally I did not see the problem but I had not removed the node modules directory before running install :-(

Adding back with [101db25](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2532/commits/101db25204d24f6a295bb0631d6f8aa90d7e9c86)

**Comment by wmathurin** [comment]
@brandonpage Locally I did not see the problem but I had not removed the node modules directory before running install :-(

Adding back with [101db25](https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2532/commits/101db25204d24f6a295bb0631d6f8aa90d7e9c86)

**Pull request merged**